torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -731,6 +745,56 @@ TensorView* clamp(TensorView* in, Val* min_val, Val* max_val) {
   return clamp(in->as<Val>(), min_val, max_val)->as<TensorView>();
 }
 
+// sum_to operator
+
+inline bool shouldReduceSumTo(const int shape, const Val* root_id_extent) {
+  return shape == 1 && !root_id_extent->isOneInt();
+}
+
+inline bool shouldReduceSumTo(const Val* shape, const Val* root_id_extent) {
+  return shape->isOneInt() && !root_id_extent->isOneInt();
+}
+
+// Name of sum_to is different from NV fuser naming,
+// this is to align with the operator name
+template <typename T>
+TensorView* sum_to_impl(TensorView* v1, const std::vector<T>& shape) {
+  TensorView* v2 = v1;
+
+  const auto& v1_root = TensorDomain::noReductions(v1->getRootDomain());
+
+  const int64_t leading_dims = v1_root.size() - shape.size();
+
+  // Reduce left most dims without keep_dim
+  if (leading_dims) {
+    std::vector<int> outer_red_dims(leading_dims);
+    std::iota(outer_red_dims.begin(), outer_red_dims.end(), 0);
+    v2 = sum(v1, outer_red_dims);
```

Comment:
Having multiple reductions even though they're just one after the other is hard to deal with since there's no inverse of rfactor. It would be better to have both this sum and the one below be in one step, and add the explicit keepdim broadcast in this function.

Reply:
I see. Updated to use up to one sum and on broadcast only. Thanks!